src/generators/markdowndocgenerator/index.ts

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { IDocumentWidget, MimeDocument } from '@jupyterlab/docregistry';
+
+import { FileEditor, IEditorTracker } from '@jupyterlab/fileeditor';
+
+import { TableOfContentsRegistry } from '../../registry';
+
+import {
+  generateNumbering,
+  sanitizerOptions,
+  isMarkdown,
+  INotebookHeading,
+  INotebookHeadingTypes
+} from '../shared';
+
+import { IInstanceTracker, ISanitizer } from '@jupyterlab/apputils';
+
+import { MarkdownDocGeneratorOptionsManager } from './optionsmanager';
+
+import { TableOfContents } from '../../toc';
+
+import { markdownDocItemRenderer } from './itemrenderer';
+
+import { markdownDocGeneratorToolbar } from './toolbargenerator';
+
+/**
+ * Create a TOC generator for markdown files.
+ *
+ * @param tracker: A file editor tracker.
+ *
+ * @returns A TOC generator that can parse markdown files.
+ */
+export function createMarkdownGenerator(
+  tracker: IEditorTracker,
+  widget: TableOfContents
+): TableOfContentsRegistry.IGenerator<IDocumentWidget<FileEditor>> {
+  // Create a option manager to manage user settings
+  const options = new MarkdownDocGeneratorOptionsManager(widget, {
+    needNumbering: true
+  });
+  return {
+    tracker,
+    usesLatex: true,
+    options: options,
+    toolbarGenerator: () => {
+      return markdownDocGeneratorToolbar(options);
+    },
+    itemRenderer: (item: INotebookHeading) => {
```

Comment:
The non-rendered markdown doesn't need an `itemRenderer`, if I understand this correctly.

Reply:
The itemRenderer controls how the item is rendered in TOC as TOC entry, so even if the non-rendered markdown is not rendered in the doc, I have this function to handle the way it is rendered in TOC (for example, include auto-numbering or not)

Reply:
Ah, okay. I still think that both markdown renderers should take `IHeadings`, since the notebook data doesn't really apply to them.

Reply:
Yeah I totally agree with you since they do not have cells. I will definitely make some changes on it